### ISSUE TITLE
Corner cases aren't detected when returning a MemoryReader timeseries view

### ISSUE THREAD

**Issue by mnmelo** [MDAnalysis/mdanalysis#1129]
In `timeseries()` the Memory reader returns a view of the coordinate array if no selection is specified. It also returns a view if the selection is the entire universe -- which is undocumented behavior. This is detected with a simple `if asel is asel.universe.atoms`.

More complex selections that do evaluate to the whole universe are missed. For instance, a timeseries made with `ag3` in the following example will be a copy rather than a view:  
``` python
ag1 = u.atoms[:100]
ag2 = u.atoms[100:]
ag3 = ag1 + ag2
ag3 is u.atoms
>>> False
np.array_equal(ag3.indices, u.atoms.indices)
>>> True
```

Since the behavior is partly undocumented anyway I'm not sure if this is a proper bug or a feature.
The solution would seem to be the comparison of all atom indices, instead of just a `is` test. This, however, can become a heavy operation (as discussed in #1080), partly defeating the purpose of returning an array view.

Perhaps the best is to just document the behavior and leave it as is?

**Comment by orbeckst** [comment]
Does something like a "read-only view" exist in numpy? In principle I would want to guard against the view being accidentally changed. One could then also add a `copy=True` kwarg to always get a copy.

**Comment by mnmelo** [comment]
My issue wasn't so much with the view itself, but with the inconsistency with which one is returned. But I agree that returning a view might in itself be a source of problems.
